File: rules/vuln_samples/negative/python/sql_injection.py
# NEGATIVE: Safe SQL — should NOT trigger sql injection rules
import sqlite3

def get_user(user_id):
    conn = sqlite3.connect("app.db")
    cursor = conn.cursor()
    # Safe: parameterized query
    cursor.execute("SELECT * FROM users WHERE id = %s", (user_id,))
    # Safe: ORM equivalent
    cursor.execute("SELECT * FROM users WHERE id = ?", [user_id])

File: rules/vuln_samples/negative/javascript/sql_injection.js
// NEGATIVE: Safe SQL in Node.js
function safeQuery(userId) {
    // Safe: parameterized query with array
    db.query("SELECT * FROM users WHERE id = ?", [userId], callback);
    // Safe: named parameters
    pool.query("SELECT * FROM users WHERE id = $1", [userId]);
}
